src/main/scala/com/databricks/labs/overwatch/pipeline/Initializer.scala

```diff
@@ -275,25 +275,29 @@ class Initializer(config: Config) extends SparkSessionWrapper {
     if (overwatchScope.head == "all") config.setOverwatchScope(config.orderedOverwatchScope)
     else config.setOverwatchScope(validateScope(overwatchScope))
 
-    // validate token secret requirements
-    // TODO - Validate if token has access to necessary assets. Warn/Fail if not
-    if (tokenSecret.nonEmpty && !disableValidations && !config.isLocalTesting) {
-      if (tokenSecret.get.scope.isEmpty || tokenSecret.get.key.isEmpty) {
-        throw new BadConfigException(s"Secret AND Key must be provided together or neither of them. " +
-          s"Either supply both or neither.")
-      }
-      val scopeCheck = dbutils.secrets.listScopes().map(_.getName()).toArray.filter(_ == tokenSecret.get.scope)
-      if (scopeCheck.length == 0) throw new BadConfigException(s"Scope ${tokenSecret.get.scope} does not exist " +
-        s"in this workspace. Please provide a scope available and accessible to this account.")
-      val scopeName = scopeCheck.head
-
-      val keyCheck = dbutils.secrets.list(scopeName).toArray.filter(_.key == tokenSecret.get.key)
-      if (keyCheck.length == 0) throw new BadConfigException(s"Key ${tokenSecret.get.key} does not exist " +
-        s"within the provided scope: ${tokenSecret.get.scope}. Please provide a scope and key " +
-        s"available and accessible to this account.")
-
-      config.registerWorkspaceMeta(Some(TokenSecret(scopeName, keyCheck.head.key)))
-    } else config.registerWorkspaceMeta(None)
+    rawParams.tokenSecret.get match {
+      case databricksSecret: TokenSecret =>
+        // validate token secret requirements
+        // TODO - Validate if token has access to necessary assets. Warn/Fail if not
+        if (tokenSecret.nonEmpty && !disableValidations && !config.isLocalTesting) {
+          if (databricksSecret.scope.isEmpty || databricksSecret.key.isEmpty) {
```

Comment:
I am confused a little bit and maybe I've missed something but in https://github.com/scribd/overwatch/pull/1/files#diff-7c4309a6fba74925b551ba489878977bb3fa11aa15d9d6ee1186d90fab59f3f0R268 for databricks secret type we extract value for scope/key and here we validate is this scope\key exists in databricks.
Looks like something wrong

Reply:
I just kept the previous logic about Databricks key related checks if token passed is Databricks one (a bit of formatting change and you may perceive it as something new).